artifactory/commands/transferconfig/transferconfig.go

```diff
@@ -99,17 +104,26 @@ func (tcc *TransferConfigCommand) getRepoFilter() *utils.RepositoryFilter {
 func (tcc *TransferConfigCommand) Run() (err error) {
 	sourceServicesManager, err := utils.CreateServiceManager(tcc.sourceServerDetails, -1, 0, tcc.dryRun)
 	if err != nil {
-		return err
+		return
 	}
 	targetServiceManager, err := utils.CreateServiceManager(tcc.targetServerDetails, -1, 0, false)
 	if err != nil {
 		return
 	}
+	targetServiceManager, err = utils.CreateServiceManager(tcc.targetServerDetails, -1, 0, false)
+	if err != nil {
+		return
+	}
 
 	if tcc.preChecks {
 		return tcc.runPreChecks(sourceServicesManager, targetServiceManager)
 	}
 
+	if tcc.merge {
+		err = tcc.RunMergeCommand(sourceServicesManager, targetServiceManager)
```

Comment:
1. The RunMergeCommand function can be made provate, because it's used inside this package only.
2. RunMergeCommand --> doConfigMerge
3. I can see that this function, as well as other functions of this struct, are placed in another file. I suggest moving all of this code into this file. It's best to have all the functions of a single struct in the same file.